lib/bolt/catalog.rb

```diff
@@ -92,6 +93,16 @@ def generate_ast(code)
       end
     end
 
+    def validate_hiera_config(hiera_config)
```

Comment:
Is this the appropriate place to check for hiera 5 version? Error looks like: 
```
{
  "kind": "bolt/apply-error",
  "msg": "Apply failed to compile for Target('x6x9wezjmz9dbih.delivery.puppetlabs.net', {\"connect-timeout\"=>10, \"tty\"=>false, \"host-key-check\"=>false, \"private-key\"=>\"~/.ssh/id_rsa-acceptance\"}): /home/cas/working_dir/bolt/lib/bolt/catalog.rb:100:in `validate_hiera_config': Hiera 5 required. Version 8 detected. (Bolt::FileError)\n\tfrom /home/cas/working_dir/bolt/lib/bolt/catalog.rb:115:in `compile_catalog'\n\tfrom /home/cas/working_dir/bolt/libexec/bolt_catalog:32:in `<main>'\n",
  "details": {
  }
}
```
I am holding off on writing test for this because I feel like it might be incorrect.

Reply:
I think we could and should check earlier, maybe at the beginning of the `apply` call after testing arguments.

Reply:
Is `bolt/file-error` ok? Moved to `apply` call after testing args and message looks like:
```
{
  "kind": "bolt/file-error",
  "msg": "Hiera 5 required. Version 8 detected.",
  "details": {
    "path": "/home/cas/.puppetlabs/bolt/hiera.yaml"
  }
}
```

Reply:
Your test specifies `version: 8` explicitly?

I'd make it a `bolt/apply-error`, I think that makes it clear what it's related to.

Reply:
I will change to `bolt/apply-error` and write a test for it. I was just manually testing with setting `version: 8` in my hiera.yaml. I had version interpolated in the error message string. Thanks @MikaelSmith  !

Reply:
I think the most likely scenario is that `version` won't be set at all, which implies version 3.

Reply:
Ok, I will leave out the version interpolation in the error message. Current implementation will throw the error when `version` is not set. This should guard against the hiera 3 default.